### PR TITLE
More tensor bits

### PR DESCRIPTION
**Pull request by dlibenzi**

### CODE REVIEW

File: torch_xla/csrc/tensor.cpp [hunk 1]
```diff
@@ -537,18 +565,18 @@ xla::int64 XLATensor::GetNextTensorId() {
   return id_generator->fetch_add(1);
 }
 
-std::shared_ptr<XLATensor> XLATensor::add(const XLATensor& other,
+std::shared_ptr<XLATensor> XLATensor::add(XLATensor& other,
```
Comment (asuhan) [inline torch_xla/csrc/tensor.cpp]: Can't we make these `std::shared_ptr` if we need to drop the `const`? This looks weird.

Comment (asuhan) [reply torch_xla/csrc/tensor.cpp]: I disagree about the mutable alternative being worse, but ok. We can go with this.

**Comment by dlibenzi** [comment]
You prefer mutable?
I find it weird (unless in very rare occasions) that the state of an object changes upon const reference passing.
Current we have Current*() and Get*() APIs, for device data, IR node and tensor data.
And the Get*() are the one which "must get" a given entity. So changing the state of the object is expected.

**Comment by asuhan** [comment]
I don't really "prefer" it, but it's the lesser evil here. If the methods are at least logically `const`, but they happen to mutate some internal state or cache, I'm usually fine with making that mutable.

**Pull request merged**